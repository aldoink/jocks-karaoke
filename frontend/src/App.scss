@import "scss/variables";

body {
  background-color: $rangers-blue;
}

input, button {
  font-family: Roboto, sans-serif;
  word-break: break-word;
}

#nav {
  position: sticky;
  position: -webkit-sticky;
  top: 0;
  display: flex;
  justify-content: space-between;
  border-bottom: $white-border-style;
  background-color: $rangers-blue;

  @media screen and (min-width: 769px) {
    border-left: $white-border-style;
    border-right: $white-border-style;
    margin: 0 2rem;
  }

  .search-field-container {
    display: flex;
    flex-grow: 1;
    padding: 1rem;

    .search-field {
      display: flex;
      border-bottom: $white-border-style;
      transition: 0.5s;
      width: 100%;

      &:focus-within {
        border-bottom: 1px solid red;
      }

      input {
        flex-grow: 1;
        min-width: 0;
        padding: 0.5rem;
        background-color: transparent;
        border: none;
        color: white;
        font-size: 16px;
        caret-color: white;

        &:focus {
          outline: none;
        }

        &::placeholder {
          color: white;
          opacity: 1;
        }
      }

      svg {
        fill: white;
        height: 1rem;
        width: 1rem;
        align-self: center;
      }
    }
  }
}

.app-container {
  max-width: 80rem;
  margin-left: auto;
  margin-right: auto;
}

.content-container {
  padding: 0 0.5rem;
  text-align: center;

  @media screen and (min-width: 768px) {
    padding: 0 2rem;
  }

  button {
    position: sticky;
    position: -webkit-sticky;
    right: 0;
    top: 0;
  }
}

.logo-container {
  display: flex;
  justify-content: center;
  margin: 2rem 0;

  .logo {
    height: 30vmin;
    pointer-events: none;
    justify-self: end;
  }

}

.table-container {
  width: 100%;
  box-sizing: border-box;
  border: 1px solid red;
  padding: 4px;
  margin: auto;
}

