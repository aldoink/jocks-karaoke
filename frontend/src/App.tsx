import React, {useEffect, useRef, useState} from 'react';
import logo from './assets/JocksKaraoke.jpeg';
import './App.scss';
import useDebounce from "./UseDebounce";
import {SongService} from "./services/SongService";
import {ReactComponent as SearchIcon} from './assets/search-icon.svg'
import {HamburgerMenu} from "./components/HamburgerMenu";
import {Table} from "./components/Table";

interface Song {
    artist: string,
    location: string,
    title: string
}

interface AppProps {
    songService: SongService
}

function App({songService}: AppProps) {
    const searchInputRef: React.RefObject<HTMLInputElement> = useRef(null)

    const [songList, setSongList] = useState([]);
    const [searchTerm, setSearchTerm] = useState("");
    const debouncedSearchTerm: string = useDebounce(searchTerm, 300);

    useEffect(() => {
        const search = async () => {
            const songList = await songService.search(debouncedSearchTerm);
            setSongList(songList?.filter((entry: Song) => {
                return entry.title.toLowerCase().includes(debouncedSearchTerm)
                    || entry.artist.toLowerCase().includes(debouncedSearchTerm)
                    || entry.location.toLowerCase().includes(debouncedSearchTerm)
            }))
        }
        search();
    }, [debouncedSearchTerm, songService]);

    const handleSearch = (event: React.ChangeEvent<HTMLInputElement>) => setSearchTerm(event?.currentTarget?.value.toLowerCase())

    return (<div className="app-container">
            <div id="nav">
                <div className="search-field-container">
                    <div className="search-field">
                        <input type="text"
                               placeholder="Search"
                               onChange={handleSearch}
                               ref={searchInputRef}/>
                        <SearchIcon/>
                    </div>
                </div>
                <HamburgerMenu/>
            </div>
            <div className="content-container">
                <header className="logo-container">
                    <img src={logo} className="logo" alt="logo"/>
                </header>
                <Table songList={songList}/>
            </div>
        </div>
    );

}

export default App;
