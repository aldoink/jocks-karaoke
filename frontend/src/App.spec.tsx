import {render} from "@testing-library/react";
import App from "./App";

describe('App', () => {
    it('renders correctly', async () => {
        expect(render(<App/>).asFragment()).toMatchSnapshot();
    });
});
