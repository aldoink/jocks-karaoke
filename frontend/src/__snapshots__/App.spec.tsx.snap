// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App loads songs and displays them in the table 1`] = `
<DocumentFragment>
  <div
    class="app-container"
  >
    <div
      id="nav"
    >
      <div
        class="search-field-container"
      >
        <div
          class="search-field"
        >
          <input
            placeholder="Search"
            type="text"
          />
          <svg>
            search-icon.svg
          </svg>
        </div>
      </div>
      <div
        class=""
        data-testid="hamburger-container"
        id="hamburger-container"
      >
        <button
          class="hamburger hamburger--squeeze "
          data-testid="hamburger-button"
          type="button"
        >
          <span
            class="hamburger-box"
          >
            <span
              class="hamburger-inner"
            />
          </span>
        </button>
      </div>
      <div
        class="menu-drawer"
        data-testid="menu-drawer"
        style="max-height: 0px; visibility: hidden;"
      >
        <div
          class="menu-item"
        >
          Login
        </div>
      </div>
    </div>
    <div
      class="content-container"
    >
      <header
        class="logo-container"
      >
        <img
          alt="logo"
          class="logo"
          src="JocksKaraoke.jpeg"
        />
      </header>
      <div
        class="table-container"
      >
        <div
          class="table-cell bold"
        >
          Song Title
        </div>
        <div
          class="table-cell bold"
        >
          Artist
        </div>
        <div
          class="table-cell bold"
        >
          Location
        </div>
        <div
          class="table-cell"
        >
          Immigrant Song
        </div>
        <div
          class="table-cell"
        >
          Led Zeppelin
        </div>
        <div
          class="table-cell"
        >
          GP001
        </div>
        <div
          class="table-cell"
        >
          Don't stop me now
        </div>
        <div
          class="table-cell"
        >
          Queen
        </div>
        <div
          class="table-cell"
        >
          GP002
        </div>
        <div
          class="table-cell"
        >
          Damn it feels good to be a gangster
        </div>
        <div
          class="table-cell"
        >
          Geto Boys
        </div>
        <div
          class="table-cell"
        >
          GP003
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
